docs/design/2020-10-20-support-flushback-table-to-tso.md

```diff
@@ -0,0 +1,101 @@
+# Proposal: Support flushback table to specific TSO
+
+- Author(s):     [yangfei](https://github.com/amyangfei)
+- Last updated:  2020-10-20
+- Discussion at: https://github.com/pingcap/tidb/issues/20302
+
+## Abstract
+
+TiDB supports reverting table data back to an arbitrary time(later than GC safe time).
+
+## Background
+
+[TiCDC](https://github.com/pingcap/ticdc) is a change data capture for TiDB, it supports to replicate change data from upstream TiDB to downstream TiDB. TiCDC supports a [tso-map feature](https://github.com/pingcap/ticdc/pull/932), which means TiCDC can replicate data to a consistent state periodically (under the conditions that upstream TiDB is available), and persists a tso map which records a consistent state between upstream and downstream. Based on that premise, when upstream meets disaster, if downstream TiDB can recover data to the latest TSO in tso-map, we could get a latest consistent state compared to upstream.
+
+We would like TiDB to provide the table flashback feature, it can be used not only in TiCDC consistent replication, but also in many data recovery scenarios, such as recovery table to eliminate recent misoperation.
+
+## Proposal
+
+### Syntax
+
+Reuse the statement syntax of existing `FLASHBACK` but add the `TIMESTAMP` keyword.
+
+```sql
+FLASHBACK TABLE tbl TIMESTAMP 420264149562425345;
```

Comment:
It's better to use the `AS OF TIMESTAMP` syntax to be compatible with time travel queries.